src/main/java/be/seeseemelk/mockbukkit/inventory/InventoryMock.java

```diff
@@ -234,14 +245,14 @@ public ItemStack addItem(@NotNull ItemStack item)
 	{
 		final int itemMaxStackSize = Math.min(item.getMaxStackSize(), this.maxStackSize);
 		item = item.clone();
-		for (int i = 0; i < items.length; i++)
+		for (int i = 0; i < this.getSize(); i++)
 		{
 			ItemStack oItem = items[i];
 			if (oItem == null)
 			{
 				int toAdd = Math.min(item.getAmount(), itemMaxStackSize);
-				items[i] = item.clone();
-				items[i].setAmount(toAdd);
+				this.setItem(i, item);
+				this.getItem(i).setAmount(toAdd);
```

Comment:
I was wrong with my initial comment. It seems like adding to an inventory can in fact also add it to the result slot. In this case `items.length==9` (matrix) and `getSize()==10` (workbench mock overrides and accounts the result slot). Subsequently calling `setItem` is necessary instead of directly assigning it to the array, because adding to the result slot has to be possible as well.